### ISSUE TITLE
[Security] CSRF token validation failure in functional tests

### ISSUE THREAD

**Issue by mmarchois** [symfony/symfony#50209]
### Symfony version(s) affected

6.2.8

### Description

Since the upgrade to version 6.2.8, the validation of csrf tokens in our functional tests does not work anymore. All my tests throw the exception "Invalid CSRF token".

### How to reproduce

The controller i want to test :
```php
// ...
public function __invoke(Request $request): Response
{
        // ...
        $csrfToken = new CsrfToken('token-name', $request->request->get('token'));
        if (!$this->csrfTokenManager->isTokenValid($csrfToken)) {
            throw new BadRequestHttpException('Invalid CSRF token');
        }
}
```

The functional test :
```php
// ...
use SessionHelper;

public function testDelete(): void
{
    $client = $this->loginUser($user) ;
    $client->request('DELETE', 'my-url', [
        'token' => $this->generateCsrfToken($client, 'token-name'),
    ]) ;
}
```

Token generation on the test side : 
```php
trait SessionHelper
{
    public function getSession(KernelBrowser $client) : Session
    {
        $cookie = $client->getCookieJar()->get('MOCKSESSID') ;

        // create a new session object
        $container = static::getContainer() ;
        $sessionSavePath = $container->getParameter('session.save_path') ;
        $sessionStorage = new MockFileSessionStorage($sessionSavePath) ;
        $session = new Session($sessionStorage) ;
        $session->setId($cookie->getValue()) ;
        $session->start() ;

        return $session ;
    }

    public function generateCsrfToken(KernelBrowser $client, string $tokenId) : string
    {
        $session = $this->getSession($client) ;
        $container = static::getContainer() ;
        $tokenGenerator = $container->get('security.csrf.token_generator') ;
        $csrfToken = $tokenGenerator->generateToken() ;
        $session->set(SessionTokenStorage::SESSION_NAMESPACE . "/{$tokenId}", $csrfToken) ;
        $session->save() ;

        return $csrfToken ;
    }
}
```
Note that this code worked in 6.2.7.

**Comment by krciga22** [comment]
Hi @mmarchois, I've taken a look at this issue and was able to recreate it on my system with two separate projects for 6.2.7 and 6.2.8. It looks like this issue is occurring in 6.2.8 due to a commit from Issue #49745. This commit sets/overwrites the session save_path config and container parameter to null if you don't specify a session handler_id in your framework config yaml. Do you currently have a session handler_id such as 'session.handler.native' or 'session.handler.native_file' defined in your framework config yaml? I tested both of these handler_ids and the test case appears to pass for both. I'm not sure why save_path=null is affecting the test case, but I'll take a deeper look and see what I can find.

[commit e23be5834812b00b321f8cfc3a968b20546aa6f7](https://github.com/symfony/symfony/pull/49745/commits/e23be5834812b00b321f8cfc3a968b20546aa6f7#diff-e4f52d5033296421d02e360c8355aa1931bdfb0c12579f4815ea02b724322c0cR1181)

```php
if (null === $config['handler_id']) {
    $config['save_path'] = null;
    $container->setAlias('session.handler', 'session.handler.native');
} else { 
    ...
}
```

```yaml
framework:
    session:
        handler_id: 'session.handler.native'
```

**Comment by krciga22** [comment]
After looking into this some more, it looks like the test case creates an instance of MockFileSessionStorage which has a save_path of '%kernel.cache_dir%/sessions' in 6.2.7, but then has a save_path of null (further defaulting to sys_get_temp_dir()) in 6.2.8 due to the fix in #49745.

The controller appears to use MockFileSessionStorageFactory which has a save_path of param('kernel.cache_dir').'/sessions' in both versions.

To fix the test case, as before you can specifcy a session handler_id in your framework config yaml, or use MockFileSessionStorageFactory in the test case instead of MockFileSessionStorage.

```php
//$sessionStorage = new MockFileSessionStorage($sessionSavePath) ;
$factory = $this->getContainer()->get('session.storage.factory.mock_file');
$sessionStorage = $factory->createStorage(null);
```

**Comment by mmarchois** [comment]
@krciga22 it works ! Many thanks for your help ! :)
